src/main/java/io/gravitee/policy/oauth2/Oauth2Policy.java

```diff
@@ -278,21 +333,33 @@ private Completable introspectAndValidateAccessToken(HttpExecutionContext ctx, S
      *      error="invalid_token",
      *      error_description="The access token expired"
      */
-    private Completable sendError(HttpExecutionContext ctx, String responseKey) {
-        String headerValue = BEARER_AUTHORIZATION_TYPE + " realm=\"gravitee.io\"";
+    private Completable sendError(BaseExecutionContext ctx, int errorStatus, String responseKey, String failureMessage) {
```

Comment:
We could take the opportunity to add an enum that could hold status, key and message.